internal/provider/resource_assigned_managed_identity_test.go

```diff
@@ -0,0 +1,107 @@
+package provider
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	ttypes "gitlab.com/infor-cloud/martian-cloud/tharsis/tharsis-sdk-go/pkg/types"
+)
+
+// TestAssignedManagedIdentity tests creation, reading, updating, and deletion of an assigned managed identity resource,
+// also known as assigning and unassigning a managed identity to/from a workspace.
+func TestAssignedManagedIdentity(t *testing.T) {
+	resource.Test(t, resource.TestCase{
+		// Tharsis assigned managed identities
+		ProtoV6ProviderFactories: testAccProtoV6ProviderFactories,
+		Steps: []resource.TestStep{
+			// Create and read back an assigned managed identity.
+			{
+				Config: testSharedProviderConfiguration() + testAssignedManagedIdentityConfiguration(),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					// There are no values in an assigned managed identity that are known.
+
+					// Verify dynamic values have any value set in the state.
+					resource.TestCheckResourceAttrSet("tharsis_assigned_managed_identity.tami", "managed_identity_id"),
+					resource.TestCheckResourceAttrSet("tharsis_assigned_managed_identity.tami", "workspace_id"),
+				),
+			},
+
+			// Import state.
+			{
+				ResourceName:      "tharsis_assigned_managed_identity.tami1",
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+
+			// Update (which requires replacement) and read back.
+			{
+				Config: testAssignedManagedIdentityConfiguration(),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					// There are no values in an assigned managed identity that are known.
+
+					// Verify dynamic values have any value set in the state.
+					resource.TestCheckResourceAttrSet("tharsis_assigned_managed_identity.tami", "managed_identity_id"),
+					resource.TestCheckResourceAttrSet("tharsis_assigned_managed_identity.tami", "workspace_id"),
+				),
+			},
+
+			// Destroy should be covered automatically by TestCase.
+		},
+	})
+}
+
+func testAssignedManagedIdentityConfiguration() string {
+	return createRootGroup(testGroupPath, "this is a test root group") +
+		testAssignedMICreateWorkspace() +
+		testAssignedManagedIdentityCreateManagedIdentity() +
+		fmt.Sprintf(`
+
+resource "tharsis_assigned_managed_identity" "tami1" {
+	managed_identity_id = tharsis_managed_identity.tmi_tharsis.id
+	workspace_id = tharsis_workspace.tw.id
+}
+`,
+		)
+}
+
+func testAssignedMICreateWorkspace() string {
+	createName := "tw_name"
+	createDescription := "this is tw, a test workspace"
+	createMaxJobDuration := 1234      // must not exceed 1440
+	createTerraformVersion := "1.2.3" // must be a valid version
+	createPreventDestroyPlan := true
+
+	return fmt.Sprintf(`
+
+resource "tharsis_workspace" "tw" {
```

Comment:
These resources can be moved into the same config instead of breaking them out in different functions.